setup.py

```diff
@@ -91,19 +91,46 @@ def build_extension(self, ext: CMakeExtension):
                     "-DCMAKE_SYSTEM_PROCESSOR=ARM64",
                 ]
             else:  # X64 arch
-                if shutil.which("brew"):
+                if os.getenv("USE_BREW_LLVM") and shutil.which("brew"):
+                    brew_llvm_version = os.getenv("BREW_LLVM_VERSION")
                     llvmpath = (
-                        subprocess.check_output(["brew", "--prefix", "llvm"]).decode().strip()
+                        subprocess.run(
+                            [
+                                "brew",
+                                "--prefix",
+                                "llvm" + f"@{brew_llvm_version}" if brew_llvm_version else "",
+                            ],
+                            check=True,
+                            capture_output=True,
+                            text=True,
+                        )
+                        .stdout
+                        .strip()
                     )
                 else:
                     llvmpath = shutil.which("clang++")
                     llvmpath = Path(llvmpath).parent.parent
+
+                libomp_path = (
+                    subprocess.run(
+                        [
+                            "brew",
+                            "--prefix",
+                            "libomp",
+                        ],
+                        check=False,
+                        capture_output=True,
+                        text=True,
+                    )
+                    .stdout
+                    .strip()
+                )
```

Comment:
This will cause issues on conda-forge, because `brew` is not installed.
I suggest conditioning `ENABLE_OPENMP` on `USE_OMP`. Then change 
```suggestion
                libomp_path = (
                    (
                        subprocess.run(
                            [
                                "brew",
                                "--prefix",
                                "libomp",
                            ],
                            check=False,
                            capture_output=True,
                            text=True,
                        ).stdout.strip()
                    )
                    if shutil.which("brew")
                    else None
                )
```
Finally, if `libomp_path` is not `None`, let
```
configure_args += [f"-DOpenMP_ROOT={libomp_path}/"]
```

Reply:
I moved this around a little more to more aptly keep things contained. While I am not a fan of the current highly nested if-else structure, it will unblock the other builds for now.